classes/Pods.php

```diff
@@ -890,94 +890,134 @@ public function field( $name, $single = null, $raw = false ) {
+		$permalink_fields = array(
 			'_link',
 			'detail_url',
-		), true ) || ( in_array( $params->name, array(
 			'permalink',
 			'the_permalink',
-		), true ) && in_array( $this->pod_data['type'], array(
+		);
+
+		$wp_object_types = array(
 			'post_type',
 			'taxonomy',
 			'media',
 			'user',
 			'comment',
-		), true ) ) ) {
+		);
+
+		/** @var string $pod_type The pod object type. */
+		$pod_type = pods_v( 'type', $this->pod_data, '' );
+
+		$is_wp_object = in_array( $pod_type, $wp_object_types, true );
+
+		if ( $is_wp_object && in_array( $params->name, $permalink_fields, true ) ) {
 			if ( 0 < strlen( $this->detail_page ) ) {
 				$value = get_home_url() . '/' . $this->do_magic_tags( $this->detail_page );
-			} elseif ( in_array( $this->pod_data['type'], array( 'post_type', 'media' ), true ) ) {
-				$value = get_permalink( $this->id() );
-			} elseif ( 'taxonomy' === $this->pod_data['type'] ) {
-				$value = get_term_link( $this->id(), $this->pod_data['name'] );
-			} elseif ( 'user' === $this->pod_data['type'] ) {
-				$value = get_author_posts_url( $this->id() );
-			} elseif ( 'comment' === $this->pod_data['type'] ) {
-				$value = get_comment_link( $this->id() );
+			} else {
+				switch ( $pod_type ) {
+					case 'post_type':
+					case 'media':
+						$value = get_permalink( $this->id() );
+						break;
+					case 'taxonomy':
+						$value = get_term_link( $this->id(), $this->pod_data['name'] );
+						break;
+					case 'user':
+						$value = get_author_posts_url( $this->id() );
+						break;
+					case 'comment':
+						$value = get_comment_link( $this->id() );
+						break;
+				}
 			}
 		}
 
-		$field_data      = false;
-		$last_field_data = false;
-		$field_type      = false;
-
-		$first_field = explode( '.', $params->name );
-		$first_field = $first_field[0];
+		/**
+		 * @var string $field_source    Regular field or object field.
+		 * @var array  $field_data      The field data.
+		 * @var string $field_type      The field type.
+		 * @var array  $field_options   The field options.
+		 * @var array  $traverse_fields All the traversal field names.
+		 * @var string $first_field     The name of the fieds without the traversal names from $params->name.
+		 * @var array  $last_field_data The field data used in traversal loop.
+		 */
 
+		$is_field_set       = isset( $this->fields[ $params->name ] );
+		$is_tableless_field = false;
+		$field_source       = '';
+		$field_data         = ( $is_field_set ) ? $this->fields[ $params->name ] : array();
+		$field_options      = array();
+		$field_type         = '';
+		$traverse_fields    = explode( '.', $params->name );
+		$first_field        = $traverse_fields[0];
+		$last_field_data    = null;
+
+		// Get the field type and it's data.
 		if ( isset( $this->fields[ $first_field ] ) ) {
-			$field_data = $this->fields[ $first_field ];
-			$field_type = 'field';
+			$field_data         = $this->fields[ $first_field ];
+			$field_source       = 'field';
```

Comment:
:warning: Equals sign not aligned with surrounding assignments; expected 1 space but found 7 spaces

<sub>hash: 3bf4070bcda7576499d2950d5a7fad15</sub>